engines/gob/surface.cpp

```diff
@@ -917,47 +1074,121 @@ ImageType Surface::identifyImage(Common::SeekableReadStream &stream) {
 	// Check width, height and bpp for sane values
 	if ((width == 0) || (height == 0) || (bpp == 0))
 		return kImageTypeNone;
-	if ((width > 800) || (height > 600))
-		return kImageTypeNone;
 	if ((bpp != 8) && (bpp != 16) && (bpp != 24) && (bpp != 32))
 		return kImageTypeNone;
 
 	// This might be a TGA
 	return kImageTypeTGA;
 }
 
-
-bool Surface::loadTGA(Common::SeekableReadStream &stream) {
-	warning("TODO: Surface::loadTGA()");
-	return false;
-}
-
-bool Surface::loadIFF(Common::SeekableReadStream &stream) {
-	Image::IFFDecoder decoder;
+bool Surface::loadImage(Image::ImageDecoder &decoder, Common::SeekableReadStream &stream, int16 left, int16 top, int16 right, int16 bottom,
+						int16 x, int16 y, int16 transp, Graphics::PixelFormat format) {
 	decoder.loadStream(stream);
 
 	if (!decoder.getSurface())
 		return false;
 
-	resize(decoder.getSurface()->w, decoder.getSurface()->h);
-	memcpy(_vidMem, decoder.getSurface()->getPixels(), decoder.getSurface()->w * decoder.getSurface()->h);
+	const Graphics::Surface *st = decoder.getSurface();
+	bool needConversion = decoder.getSurface()->format.bytesPerPixel > 1 && decoder.getSurface()->format != format;
+	if (needConversion) {
+		Graphics::Surface *st2 = st->convertTo(format);
+		// Force the pixel value 0 (used by the engine as the special transparent color) to be still mapped to 0 in the new format
```

Comment:
If the pixel format of the converted surface doesn't have any alpha bits, you should be able to skip this step.